app/src/main/java/com/github/se/polyfit/ui/components/AddIngredientPopup.kt

```diff
@@ -0,0 +1,197 @@
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import com.github.se.polyfit.ui.theme.*
+import com.github.se.polyfit.ui.utils.removeLeadingZerosAndNonDigits
+
+// Constants
+// Add to list for new nutrition fields
+val NUTRITION_LABELS = listOf("Serving Size", "Calories", "Carbs", "Fat", "Protein")
+val NUTRITION_UNITS = listOf("g", "kcal", "g", "g", "g")
+// Temporary list of available ingredient to search
+val TMP_AVAILABLE_INGREDIENT = listOf("Apple", "Banana", "Carrot", "Date", "Eggplant", "apes")
+
+@Composable
+fun NutritionFacts() {
+
+  val nutritionLabels = NUTRITION_LABELS
+  val nutritionUnit = NUTRITION_UNITS
+
+  // TODO: Integrate backend data fields here or pass in nutrition viewmodel
+  val nutritionSize = remember { mutableStateListOf("0", "0", "0", "0", "0") }
+
+  val nutritionLabelCount = nutritionLabels.size - 1
+
+  (0..nutritionLabelCount).forEach { index ->
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier =
+            Modifier.testTag("NutritionInfoContainer " + nutritionLabels[index])
+                .fillMaxWidth()
+                .padding(start = 20.dp, end = 0.dp)) {
+          Text(
+              text = nutritionLabels[index],
+              color = SecondaryGrey,
+              style = TextStyle(fontSize = 18.sp),
+              modifier = Modifier.testTag("NutritionLabel " + nutritionLabels[index]).weight(1.5f))
+
+          TextField(
+              value = nutritionSize[index],
+              onValueChange = { newValue ->
+                nutritionSize[index] = removeLeadingZerosAndNonDigits(newValue)
+              },
+              modifier =
+                  Modifier.testTag("NutritionSizeInput " + nutritionLabels[index]).weight(0.5f),
+              singleLine = true,
+              colors =
+                  TextFieldDefaults.colors(
+                      focusedIndicatorColor = PrimaryPurple,
+                      unfocusedIndicatorColor = SecondaryGrey,
+                      focusedContainerColor = Color.Transparent,
+                      unfocusedContainerColor = Color.Transparent))
+
+          Text(
+              text = nutritionUnit[index],
+              style = TextStyle(fontSize = 18.sp),
+              color = SecondaryGrey,
+              modifier =
+                  Modifier.testTag("NutritionUnit " + nutritionLabels[index])
+                      .weight(0.4f)
+                      .padding(start = 8.dp))
+        }
+    Spacer(modifier = Modifier.height(10.dp))
+  }
+}
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchIngredients() {
```

Comment:
I noticed a weird behavior when testing the SearchIngredients feature. When I click on the search bar, its shape changes from an oval to a rectangle. This might be related to how the Material 3 component is implemented. I just want to make sure that you are also experiencing this on your end.

Reply:
Yes it is just the default behavior of `SearchBar`, I thought after clicking, it kinda looks like it expands when it turns into a large rectangle, so I left it in and recognize it as a feature not a bug XD